src/plot_utils.py

```diff
@@ -0,0 +1,140 @@
+import matplotlib.pyplot as plt 
+import numpy as np
+import os
+import pandas as pd
+import torch as th
+
+def plot_losses(losses, val_losses, model_name):
+    epochs = np.arange(len(losses))
+
+    plt.subplot(1, 2, 1)
+    plt.title(f"{model_name} - Power Flow Training Learning Curve")
+    plt.plot(epochs, losses, label="Training Loss")
+    plt.legend()
+    plt.xlabel("Epochs")
+    plt.ylabel("MSE")
+
+    plt.subplot(1, 2, 2)
+    plt.title(f"{model_name} - Power Flow Validation Learning Curve")
+    plt.plot(epochs, val_losses, label="Validation Loss")
+    plt.legend()
+    plt.xlabel("Epochs")
+    plt.ylabel("MSE")
+
+    plt.tight_layout()
+    plt.show()
+
+
+def distance_plot(model, batch):
+    out = model(batch)
+    distance_loss,len = get_distance_loss(out,batch.y,batch)
+    plt.bar(list(range(0,len)), distance_loss, color ='maroon')
+    plt.title("Error with distance from the generator")
+    plt.ylabel("Error")
+    plt.xticks(range(0,len))
+    plt.xlabel("Nodes away from the generator the node was located")
+    plt.show()
+
+    # change directory to root of project
+    os.chdir(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    if not os.path.exists("plots"):
+        os.mkdir("plots")
+
+    timestamp = pd.Timestamp.now().strftime("%Y-%m-%d %H:%M")
+
+    model_name = "distance_plot" + "_" + model.class_name + "_" + str(timestamp)
+    plt.savefig(f"plots/{model_name}.png")
+
+
+def get_distance_loss(out,labels,data):
+    res = [0]
+    norm = [0]
+    distances = get_distance_from_generator(data)
+    for i, dis in enumerate(distances):
+        if dis != -1:
+            if dis > len(res)-1:
+                res = res + [0]*(dis-len(res) +1)
+                norm = norm + [0]*(dis-len(norm) +1)
+            norm[dis] += 1
+            res[dis] += th.sum(th.abs(out[i]-labels[i])).item()
+    for i in range(len(res)):
+        if norm[i] != 0:
+            res[i] = res[i]/norm[i]
+    return res, len(res)
```

Comment:
maybe some comments or a more explicit variable naming makes the code easier to read

Reply:
Added documentation